spec/controllers/queue_items_controller_spec.rb

```diff
@@ -103,9 +97,104 @@
     it "affects the position of subsequent items in the user's queue" do
       expect(QueueItem.last.position).to eq 2
     end
-    
+
     it 'redirects to the my queue page' do
       expect(response).to redirect_to my_queue_path
     end
   end
+
+  describe 'PATCH update' do
+    context "with valid updates to current user's queue" do
+      let(:user) { User.find test_login }
+      let!(:queue) { render_queue(user, 5) }
+
+      it 'redirects to the my queue page' do
+        queue_params = queue.map do |item|
+          { id: item.id, position: item.position }
+        end
+        patch :update, queue: queue_params
+        expect(response).to redirect_to my_queue_path
+      end
+
+      it "updates the positions of items in the queue, if positions were modified" do
+        queue[0][:position] = 4
+        queue[3][:position] = 1
+
+        queue_params = queue.map do |item|
+          { id: item.id, position: item.position }
+        end
+        patch :update, queue: queue_params
+        expect(QueueItem.find(queue[0].id).position).to eq 4
+      end
+
+      it "reoders the users queue, according to position" do
+        queue[0][:position] = 4
+        queue[3][:position] = 1
+
+        queue_params = queue.map do |item|
+          { id: item.id, position: item.position }
+        end
+        patch :update, queue: queue_params
+        expect(user.queue_items[0]).to eq queue[3]
```

Comment:
It is a little bit difficult to know for sure because of your abstractions, but there is a good chance you should use `user.queue_items[0].reload` to make sure you have the latest object data from the database.